moonraker/components/zeroconf.py

```diff
@@ -0,0 +1,64 @@
+# MQTT client implementation for Moonraker
+#
+# Copyright (C) 2021  Eric Callahan <arksine.code@gmail.com>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+from __future__ import annotations
+import socket
+import asyncio
+
+from zeroconf import IPVersion
+from zeroconf.asyncio import AsyncServiceInfo, AsyncZeroconf
+
+from typing import TYPE_CHECKING, Callable, Coroutine, List, Optional
+if TYPE_CHECKING:
+    from app import APIDefinition
+    from confighelper import ConfigHelper
+    FlexCallback = Callable[[bytes], Optional[Coroutine]]
+    RPCCallback = Callable[..., Coroutine]
+
+class AsyncRunner:
+    def __init__(self, ip_version: IPVersion) -> None:
+        self.ip_version = ip_version
+        self.aiozc: Optional[AsyncZeroconf] = None
+
+    async def register_services(self, infos: List[AsyncServiceInfo]) -> None:
+        self.aiozc = AsyncZeroconf(ip_version=self.ip_version)
+        tasks = [self.aiozc.async_register_service(info) for info in infos]
+        background_tasks = await asyncio.gather(*tasks)
+        await asyncio.gather(*background_tasks)
+        while True:
+            await asyncio.sleep(1)
```

Comment:
I'm not sure what the purpose is of the while loop here.

Reply:
Heh yes... sorry this was a quick side task in a bigger project, I'll look it over and test today then update the MR.